### PR TITLE
Fix null checks before Android Q

### PR DESCRIPTION
**Pull request by NotWoods**
SDK 29 adds many more `@Nullable` and `@NotNull` checks, which causes parts of our code to break since they assume nullability one way or another. I pulled some of the fixes I needed to get compling to work into their own branch here.

### CODE REVIEW

File: components/browser/domains/src/main/java/mozilla/components/browser/domains/Domains.kt [hunk 1]
```diff
@@ -46,9 +48,9 @@ object Domains {
         val availableDomains = LinkedHashSet<String>()
         val assetManager = context.assets
         val domains = try {
-            assetManager.list("domains")
+            assetManager.list("domains") ?: emptyArray<String>()
```
Comment (dector) [inline components/browser/domains/src/main/java/mozilla/components/browser/domains/Domains.kt]: nit: Alternatively can be something like

```kotlin
val domains: Array<String> = try {
    assetManager.list("domains")
} catch (e: IOException) {
    null
} ?: emptyArray()
```

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/MultiButtonDialogFragment.kt [hunk 2]
```diff
@@ -29,11 +29,11 @@ internal class MultiButtonDialogFragment : PromptDialogFragment() {
      */
     internal val hasShownManyDialogs: Boolean by lazy { safeArguments.getBoolean(KEY_MANY_ALERTS) }
 
-    internal val positiveButtonTitle: String by lazy { safeArguments.getString(KEY_POSITIVE_BUTTON_TITLE) }
+    internal val positiveButtonTitle: String? by lazy { safeArguments.getString(KEY_POSITIVE_BUTTON_TITLE) }
```
Comment (dector) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/MultiButtonDialogFragment.kt]: Just wondering if it's possible to keep this values as non-nullable with:

```kotlin
by lazy { safeArguments.getString(KEY_POSITIVE_BUTTON_TITLE) ?: "" }
```

File: components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt [hunk 3]
```diff
@@ -77,7 +77,7 @@ class SafeIntent(val unsafe: Intent) {
 
     fun <T : Parcelable> getParcelableArrayListExtra(name: String): ArrayList<T>? {
         return safeAccess {
-            val value: ArrayList<T> = unsafe.getParcelableArrayListExtra(name)
+            val value: ArrayList<T>? = unsafe.getParcelableArrayListExtra(name)
```
Comment (dector) [inline components/support/utils/src/main/java/mozilla/components/support/utils/SafeIntent.kt]: nit: can it be simplified to

```kotlin
fun <T : Parcelable> getParcelableArrayListExtra(name: String): ArrayList<T>? =
    safeAccess { unsafe.getParcelableArrayListExtra(name) }
```

**Comment by dector** [comment]
Cool! Do we have an (epic) issue for raising target/compile sdk up to 29?

**Comment by NotWoods** [comment]
new meta issue: #3481

**Comment by pocmo** [comment]
There seem to be some test failures and a merge conflict.

**Pull request merged**